Signal/src/UIView+OWS.m

```diff
@@ -299,6 +299,32 @@ - (NSLayoutConstraint *)autoPinTrailingToSuperviewWithMargin:(CGFloat)margin
     }
 }
 
+- (NSLayoutConstraint *)autoPinBottomToSuperviewWithMargin:(CGFloat)margin
+{
+    if (@available(iOS 9.0, *)) {
+        NSLayoutConstraint *constraint =
+        [self.bottomAnchor constraintEqualToAnchor:self.superview.layoutMarginsGuide.bottomAnchor
+                                          constant:-margin];
```

Comment:
What's the difference in behavior here?

Reply:
Similar to the other `autoPin*ToSuperViewMargin` methods you'd added previously, this uses the layoutMargins Guide anchors rather than the edge.

The key is that iPhoneX adjusts the layoutMarginsGuide to reflect the areas that are not obscured by the home button.

Specifically, this method is used by the InputToolbar to make sure it appears above the home button on iPhoneX, and on other devices it has no real effect, since the layoutMarginsGuide.bottomAnchor is 0.